core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -702,6 +641,87 @@ private[spark] class MemoryStore(
   }
 }
 
+private trait ValuesBuilder[T] {
+  def preciseSize: Long
+  def build(): MemoryEntry[T]
+}
+
+private trait ValuesHolder[T] {
+  def storeValue(value: T): Unit
+  def estimatedSize(): Long
+  def getBuilder(): ValuesBuilder[T]
```

Comment:
add a comment to say that, after `getBuilder` is called, this `ValuesHolder` becomes invalid.